WeatherToday/Sources/Presentation/Modules/WeatherViewModel.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  WeatherViewModel.swift
+//  WeatherToday
+//
+//  Created by Taher on 7/10/22.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Foundation
+import RxCocoa
+import RxSwift
+
+protocol WeatherViewModelInput {
+
+    func didFinishTyping(cityName: String)
+    func viewDidLoad()
+}
+
+protocol WeatherViewModelOutput {
+
+    var temperature: Signal<String> { get }
+    var humidity: Signal<String> { get }
+    var cityTextFieldPlaceholderText: String { get }
+}
+
+protocol WeatherViewModelProtocol {
+
+    var input: WeatherViewModelInput { get }
+    var output: WeatherViewModelOutput { get }
+}
+
+final class WeatherViewModel: WeatherViewModelProtocol, WeatherViewModelOutput, WeatherViewModelInput {
+
+    var input: WeatherViewModelInput { self }
+    var output: WeatherViewModelOutput { self }
+
+    var temperature: Signal<String> = .empty()
+    var humidity: Signal<String> = .empty()
+    var cityTextFieldPlaceholderText: String {
+        LocalizedString.city_text_field_placeholder_text
+    }
+
+    private let weatherUseCase: WeatherUseCaseProtocol
+    private var temperatureTextRelay: PublishRelay<String> = .init()
+    private var humidityTextRelay: PublishRelay<String> = .init()
+    private let disposeBag = DisposeBag()
+
+    init(weatherUseCase: WeatherUseCaseProtocol) {
+        self.weatherUseCase = weatherUseCase
+        temperature = temperatureTextRelay.asSignal()
+        humidity = humidityTextRelay.asSignal()
+    }
+
+    func didFinishTyping(cityName: String) {
+        weatherUseCase.getWeather(forCity: cityName).subscribe {[weak self] event in
```

Comment:
```suggestion
        weatherUseCase.getWeather(forCity: cityName).subscribe(with: self) { owner, event in
```

Reply:
resolved in this [commit](https://github.com/nimblehq/ci-cd-sample-ios/pull/20/commits/d19faa8fbaec66b5f882dd4d38f85f4a4ca738c1)